src/app.rs

```diff
@@ -62,7 +62,7 @@ pub struct Application {
     pub require_healthy: Option<bool>,
     pub config: ApplicationConfig,
     pub signals: SignalPair,
-    pub openssl_legacy_provider: Option<Provider>,
+    pub openssl_providers: Option<Vec<Provider>>,
```

Comment:
I'd suggest we refactor this in https://github.com/vectordotdev/vector/pull/18261 to not be an `Option` and just always load the default provider if none are specified.

Reply:
Taking note of this comment for #18261 ✍️